### DISCUSSION TITLE
HomeSpan ad Nano RP2040 Connect

### DISCUSSION THREAD

**Discussion by MatSquikki** [HomeSpan/HomeSpan#319] [General]
Hello
Is the integration of the library on "Nano RP2040 Connect" feasible?
Whereas it is equipped with a "U-blox® Nina W102" wifi chip, based on ESP32. "

**Comment by HomeSpan** [comment]
I don't think this would work - the RP2040 is a Raspberry PI and even though there is an ESP32 on board for WiFi, I'm not sure how it can be accessed directly.

**Comment by fsargent** [comment]
@HomeSpan How difficult would it be to modify HomeSpan to support the RP2040? Would it be worth it to try, or more effective to have the RP2040 community build their own HomeKit library from scratch? If the former, I'll do what I can to give it a go.

**Reply by HomeSpan** [reply]
I think the core HomeKit logic in HomeSpan can be ported, but this requires a variety of crypto and other libraries that are bundled with the ESP32 and would need to be available on the RPi.  The interface logic is all standard C++ but it uses dedicated WiFi and MDNS libraries which are also not likely to be available on the RPi.  Finally, the RPi does not have the pin control and embedded peripherals contained in the ESP32, so things like PWM, RF, Pixel, control are not possible (without created a lot of custom code).

Why not use HomeBridge for the RPi?

**Reply by fsargent** [reply]
Sorry, the RP2040 isn't a Raspberry Pi running Linux, it's their microcontroller, so like ESP32. https://en.wikipedia.org/wiki/RP2040

Looks like ESPHome just added support for it. https://github.com/esphome/esphome/pull/3284

**Reply by honzakotrs** [reply]
@fsargent & @HomeSpan I'd love to see RP2040 port of HomeSpan too. Are there any news on this?

**Reply by fsargent** [reply]
I ended up using Homebridge, but I did like using homespan’
